lib/index.ts

```diff
@@ -0,0 +1,405 @@
+/**
+ * @packageDocumentation
+ * TopStats - Official Node.js client for the topstats.gg API
+ * @version 1.0.0
+ */
+
+/**
+ * Response structure for rate limit errors
+ */
+export interface RateLimitedResponse {
+  statusCode: number;
+  error: string;
+  message: string;
+  date: string;
+  expiresIn: string;
+}
+
+/**
+ * Represents detailed bot data from the API
+ */
+export interface BotData {
+  /** @deprecated */
+  certified: boolean;
+  /** Array of bot owner Discord IDs */
+  owners: string[];
+  /** Whether the bot has been deleted */
+  deleted: boolean;
+  /** Bot's Discord ID */
+  id: string;
+  /** Bot's display name */
+  name: string;
+  /** Default avatar URL */
+  def_avatar: string;
+  /** Short description of the bot */
+  short_desc: string;
+  /** @deprecated */
+  lib: string;
+  /** Bot's command prefix */
+  prefix: string;
+  /** Bot's website URL */
+  website: string;
+  /** ISO timestamp of when the bot was approved */
+  approved_at: string;
+  /** Current monthly votes */
+  monthly_votes: number;
+  /** Current server count */
+  server_count: number;
+  /** Total votes received */
+  total_votes: number;
+  /** Current shard count */
+  shard_count: number;
```

Comment:
The mix of camel case and snake case would make the experience inconsistent.
It's my understanding that the API returns snake case but the internal pieces of the library use camel case.
It may be not worth the effort but getting them to be the same throughout the library would make this much nicer. Auto conversion?